particles/Renderer.cpp

```diff
@@ -326,26 +349,22 @@ void Renderer::init(score::gfx::RenderList& renderer)
             cb.setShaderResources(it->second.srb);
             cb.setViewport(QRhiViewport(0, 0, sz.width(), sz.height()));
 
-            assert(this->m_meshBuffer);
-            assert(this->m_meshBuffer->usage().testFlag(QRhiBuffer::VertexBuffer));
+            SCORE_ASSERT(this->m_meshBuffer);
+            SCORE_ASSERT(this->m_meshBuffer->usage().testFlag(QRhiBuffer::VertexBuffer));
 
             const QRhiCommandBuffer::VertexInput bindings[]
                     = {
                 {this->m_meshBuffer, 0},
-                {this->m_meshBuffer, mesh.vertexCount*sizeof(float)},
-                {this->m_particleOffsets, 0},
-                {this->m_particleSpeeds, 0},
```

Comment:
This is another bug which made it crash here: the only thing that should go in VertexInput is what is called "attributes" in opengl: e.g. 

- the mesh data
- the per-instance particle position

for instance you can see that 

    particleOffsets = renderer.state.rhi->newBuffer(
                QRhiBuffer::Immutable, QRhiBuffer::VertexBuffer | QRhiBuffer::StorageBuffer, maxparticles * 4 * sizeof(float));
    SCORE_ASSERT(particleOffsets->create());
    particleSpeeds = renderer.state.rhi->newBuffer(
                QRhiBuffer::Immutable, QRhiBuffer::StorageBuffer, maxparticles * 4 * sizeof(float));
                
here ^ particleOffsets was created as QRhiBuffer::VertexBuffer while particleSpeeds only as StorageBuffer. This means that they will actually go to different places in the GPU RAM ; if something not marked as VertexBuffer goes into a vertex attribute... crash.

Same for the ParticleControls which are just an UBO ; UBOs are never vertex inputs, they're small additional parameters which have their own space in GPU ram.